dev-support/hbase-personality.sh

```diff
@@ -140,7 +140,10 @@ function personality_modules
 
   clear_personality_queue
 
-  extra="-DHBasePatchProcess"
+  # At a few points, hbase modules can run their test suites in parallel
```

Comment:
Not just parallel test suites, but everything. With this change, maven will run up to 2 concurrent modules when the dependency graph permits. I imaging this will be a nice general benefit.

One thing to note though is that some plugins don't support multithreading. IIRC, surefire-reports was one such plugin, and would print warnings in the build. Looks like this patch didn't actually trigger any maven invocation, so I cannot point you at the message. A simple `mvn clean package -DskipTests` should be enough to get the message printed.

Since we use Jenkins to aggregate test results, I suggest we disable/remove surefire-reports (this was something I discussed with @busbey sometime back, but neither of us have gotten to it).

Reply:
Let me change the comment.

I tried running the mvn clean ... above to find the message you refer to but too blind to see it. Let me add change in pom so its generated here.

On disabling the surefire reporting, you want that in this commit or on a follow-on?  Is it this you are referring too?

maven-surefire-report-plugin

Reply:
Yeah that's the plugin. Weirdly, now I try but I don't see the warning message. Hmm..

Running with `-T1.0C` I get a failure in

```
[INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (check-aggregate-license) @ hbase-shaded-client ---                                                                                                           
[WARNING] Rule 0: org.apache.maven.plugins.enforcer.EvaluateBeanshell failed with message:                                                                                                                      
Couldn't evaluate condition: File license = new File("/Users/ndimiduk/repos/apache/hbase/hbase-shaded/hbase-shaded-client/target/maven-shared-archive-resources/META-INF/LICENSE");                                                                                                                                                                                                                                             
                    // Beanshell does not support try-with-resources,                                   
                    // so we must close this scanner manually                                                                                                                                                   
                    Scanner scanner = new Scanner(license);                                                                                                                                                     
                                                                                                                                                                                                                
                    while (scanner.hasNextLine()) {                                                                                                                                                             
                      if (scanner.nextLine().startsWith("ERROR:")) {                                                                                                                                            
                        scanner.close();                                                                                                                                                                        
                        return false;                                                                                                                                                                           
                      }                                                                                                                                                                                         
                    }                                                                                                                                                                                           
                    scanner.close();                                                                                                                                                                            
                    return true;
```

Reply:
Yeah, I see those WARNINGs locally too. Another issue?

Reply:
For me it's not just a warning, it breaks the build. Yes another issue, but if it consistently breaks runs using `-T`, it would need to be a blocker of this one.